### ISSUE TITLE
Specification for an integer64 type

### ISSUE THREAD

**Issue by carterkozak** [palantir/conjure#103]
We have several projects extensively using 64-bit integers which cannot be migrated to integer53 (`safelong`).

**Comment by iamdanfox** [comment]
Interestingly, [typescript 3.2](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-2.html) now has a `bigint` type which is enabled if you target `esnext` (expecting this proposal to be ratified: https://github.com/tc39/proposal-bigint).

We could emit code that uses `bigints` (which will be fast in modern browsers) and use this polyfill to ensure we don't drop old browsers: https://github.com/GoogleChromeLabs/jsbi

**Comment by cakofony** [comment]
While I don't think we can use `bigint` until we have a typescript ser/de layer, we can move forward with support for the conjure `integer64` type as long as we get the wire format correct.
The proposed `bigint` type does not include changes to json parsing, so we will need to serialize over the wire using a type that is not a json number. Much like our UUID type, we can serialize `integer64` values json strings, I propose using base-10 representation in a json string value.
When we implement the ser/de layer we can apply bigint parsing and expose more ergonomic types to typescript consumers.

Thoughts?

**Comment by sfackler** [comment]
I agree that it seems reasonable to move forward without waiting on full first-class Typescript support.

**Comment by cakofony** [comment]
RFC is pushed here: #227

**Comment by asanderson15** [comment]
@iamdanfox @carterkozak Are there any updates on TS serialization support (which seems to be the blocker, looking at the RFC PR above)?

**Comment by carterkozak** [comment]
If implemented, this would produce a string value for TS consumers (prior art: uuid, datetime, etc)

**Comment by asanderson15** [comment]
Right. The last comment was 1.5 years ago, so I guess I'm asking if this is still blocked on anything on the TS support side or anything else, or if it's only blocked on prioritization at this point.

**Comment by tabboud** [comment]
This all seems reasonable from the go-infra side of things. If we come to consensus on the RFC I would be happy to prioritize the work to get this into conjure-go.
